src/main/antlr4/au/csiro/data61/aap/elf/parsing/Ethql.g4

```diff
@@ -6,7 +6,20 @@
 grammar Ethql;
 
 document
-    : statement* EOF
+    : (connection outputFolder)? statement* EOF
+    ;
+
+// TODO: add in the above statement
+//blockchainType
+//    : KEY_SET KEY_BLOCKCHAIN STRING_LITERAL
+//    ;
+
+connection
+    : KEY_SET (KEY_IPC)? KEY_CONNECTION literal
```

Comment:
Think about adding a keyword KEY_WS to explicitly define a web sockets connection. This increases readability of the code. However if no keyword (KEY_IPC or KEY_WS) is set the default could be a web socket connection.